modules/performance-statistics-ext/src/main/java/org/apache/ignite/internal/performancestatistics/PerformanceStatisticsPrinter.java

```diff
@@ -0,0 +1,232 @@
+
+                for (String op : ops) {
+                    OperationType opType = enumIgnoreCase(op, OperationType.class);
+
+                    A.ensure(opType != null, "Unknown operation type [op=" + op + ']');
+
+                    params.ops.set(opType.id());
+                }
+            }
+            else if ("--from".equalsIgnoreCase(arg)) {
+                A.ensure(iter.hasNext(), "Expected time from");
+
+                params.from = Long.parseLong(iter.next());
+            }
+            else if ("--to".equalsIgnoreCase(arg)) {
+                A.ensure(iter.hasNext(), "Expected time to");
+
+                params.to = Long.parseLong(iter.next());
+            }
+            else if ("--caches".equalsIgnoreCase(arg)) {
+                A.ensure(iter.hasNext(), "Expected cache names");
+
+                String[] caches = iter.next().split(",");
+
+                A.ensure(caches.length > 0, "Expected at least one cache name");
+
+                params.cacheIds = Arrays.stream(caches).map(CU::cacheId).collect(Collectors.toSet());
+            }
+            else
+                throw new IllegalArgumentException("Unknown command: " + arg);
+        }
+
+        return params;
+    }
+
+    /** Prints help. */
+    private static void printHelp() {
+        String ops = Arrays.stream(OperationType.values()).map(Enum::toString).collect(joining(", "));
+
+        System.out.println("The script is used to print performance statistics files to the console or file." +
+            U.nl() + U.nl() +
+            "Usage: print-statistics.sh path_to_files [--out out_file] [--ops op_types] " +
+            "[--from startTimeFrom] [--to startTimeTo] [--caches cache_names]" + U.nl() +
+            U.nl() +
+            "  path_to_files - Performance statistics file or files directory." + U.nl() +
+            "  out_file      - Output file." + U.nl() +
+            "  op_types      - Comma separated list of operation types to filter the output." + U.nl() +
+            "  from          - The minimum operation start time to filter the output." + U.nl() +
+            "  to            - The maximum operation start time to filter the output." + U.nl() +
+            "  cache_names   - Comma separated list of cache names to filter the output." + U.nl() +
+            U.nl() +
+            "Times must be specified in the Unix time format in milliseconds." + U.nl() +
+            "List of operation types: " + ops + '.');
+
+        System.exit(0);
+    }
+
+    /** @param params Validates parameters. */
+    private static void validateParameters(Parameters params) {
+        File statFiles = new File(params.statFileOrDir);
+
+        A.ensure(statFiles.exists(), "Performance statistics file or files directory does not exists");
+
+        if (params.outFile != null) {
+            File out = new File(params.outFile);
+
+            try {
+                boolean created = out.createNewFile();
+
+                if (!created) {
```

Comment:
Can we just append to the existing file?

Reply:
Done.